sam/io.go

```diff
@@ -165,6 +324,78 @@ func processAlignmentLine(line string) Sam {
 	return curr
 }
 
+// processAlignmentLineRecycle parses a string representation of a sam file into a single Sam struct.
+func processAlignmentLineRecycle(line string, curr *Sam) {
```

Comment:
I am wondering if we can reduce code duplication between this function and the original processAlignmentLine.  Can process alignment line become a wrapper around this function that:
1) creates a new sam struct
2) passes it to this function
3) returns the sam